### PR TITLE
tests: robustify testCreate by mocking archive.fedoraproject.org

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

**Comment by KKoukiou** [comment]
Seems like /etc/hosts file is missing in some images - https://logs.cockpit-project.org/logs/pull-13848-20200404-103507-41a72492-rhel-8-2/log.html#84-2
Needs fix.

**Comment by martinpitt** [comment]
testCreate still failed a lot. On [Ubuntu](https://logs.cockpit-project.org/logs/pull-13848-20200404-144301-41a72492-ubuntu-2004/log.html#84) there is no /etc/hosts by default (and really other OSes should just drop it as well), and it also failed on [fedora-31](https://logs.cockpit-project.org/logs/pull-13848-20200404-144256-41a72492-fedora-31/log.html#84), also with [firefox](https://logs.cockpit-project.org/logs/pull-13848-20200404-144253-41a72492-fedora-31-firefox/log.html#84)

File: test/verify/machineslib.py [hunk 1]
```diff
@@ -1436,6 +1436,61 @@ def installWithErrorTest(dialog):
                 .deleteVm(dialog) \
                 .checkEnvIsEmpty()
 
+        def setupMockFileServer():
+            self.machine.upload(["verify/files/min-openssl-config.cnf", "verify/files/mock-range-server.py"], "/tmp/")
```
Comment (martinpitt) [inline test/verify/machineslib.py]: It's a bit confusing that this commit uses /tmp, and the next one immediately changed it. So the net result is that this PR uses vm_tmpdir.

File: test/verify/machineslib.py [hunk 2]
```diff
@@ -1436,6 +1436,61 @@ def installWithErrorTest(dialog):
...
+            fedora_28_xml = self.machine.execute("cat /usr/share/osinfo/os/fedoraproject.org/fedora-28.xml")
+            root = ET.fromstring(fedora_28_xml)
+            root.find('os').find('resources').find('minimum').find('ram').text = '134217750'
+            root.find('os').find('resources').find('minimum').find('storage').text = '134217750'
+            root.find('os').find('resources').find('recommended').find('ram').text = '268435500'
+            root.find('os').find('resources').find('recommended').find('storage').text = '268435500'
+            new_fedora_28_xml = ET.tostring(root)
+            self.machine.execute("echo \'{0}\' > /tmp/fedora-28.xml".format(str(new_fedora_28_xml, 'utf-8')))
+            self.machine.execute("mount -o bind /tmp/fedora-28.xml /usr/share/osinfo/os/fedoraproject.org/fedora-28.xml")
+            self.addCleanup(self.machine.execute, "umount /usr/share/osinfo/os/fedoraproject.org/fedora-28.xml || true")
```
Comment (martinpitt) [inline test/verify/machineslib.py:1492]: The `|| true` should not be necessary here. The test does not unmount the directory by itself.

**Comment by martinpitt** [comment]
Nice, much better! [fedora-31](https://logs.cockpit-project.org/logs/pull-13848-20200406-090601-ef68e6b6-fedora-31/log.html#84-2) now fails testCreate on an unexpected message:

    127.0.0.1:5901: couldn't shutdown fd: Transport endpoint is not connected

@KKoukiou : `testConfigureBeforeInstall` already whitelists that, can you please do the same to testCreate? But that can be a follow-up PR. Just restarting ubuntu-2004, which failed in a unit test.

**Comment by KKoukiou** [comment]
> Nice, much better! [fedora-31](https://logs.cockpit-project.org/logs/pull-13848-20200406-090601-ef68e6b6-fedora-31/log.html#84-2) now fails testCreate on an unexpected message:
> 
> ```
> 127.0.0.1:5901: couldn't shutdown fd: Transport endpoint is not connected
> ```
> 
> @KKoukiou : `testConfigureBeforeInstall` already whitelists that, can you please do the same to testCreate? But that can be a follow-up PR. Just restarting ubuntu-2004, which failed in a unit test.

Yes let's make it a followup in order to get the most flake fix of the distros at least asap.

**Pull request merged**